### PR TITLE
add real AMX tests to benchmark

### PR DESCRIPTION
**Pull request by ignaciosica**
Can someone trigger benchmark job?

### CODE REVIEW

File: .github/workflows/benchmark.yml [hunk 1]
```diff
@@ -62,6 +62,8 @@ jobs:
       run: BIG=2 MPS=1 python3.11 test/test_speed_v_torch.py | tee torch_speed.txt
     - name: Test tensor cores
       run: METAL=1 python3.11 test/test_linearizer.py TestLinearizer.test_tensor_cores TestLinearizer.test_tensor_cores_padded
+    - name: Test AMX tensor cores
+      run: CLANG=1 AMX=1 python3.11 test/test_linearizer.py TestLinearizer.test_tensor_cores TestLinearizer.test_tensor_cores_padded
```
Comment (chenyuxyz) [inline .github/workflows/benchmark.yml:66]: you would need `DEBUG=2` or some other ways to tell if AMX is really being used

**Comment by geohot** [comment]
Oh, do we have to use benchmark? What about the Apple runner in GitHub CI?

**Comment by ignaciosica** [comment]
> Oh, do we have to use benchmark? What about the Apple runner in GitHub CI?

I already tested that and its not supported in the apple runner

File: .github/workflows/benchmark.yml (hunk 1 above)
Comment (ignaciosica) [reply .github/workflows/benchmark.yml:66]: Sorry, I thought the tests asserted that they were triggered

Comment (chenyuxyz) [reply .github/workflows/benchmark.yml]: oh yea it asserts, i was thinking running `simple_matmul` with AMX like other gemm

**Comment by chenyuxyz** [comment]
we can benchmark this like other gemm later. 2 seconds is fine to start with, and i do see test asserts if tc is not used

**Pull request merged**